packages/flutter_tools/lib/src/base/process.dart

```diff
@@ -607,3 +609,387 @@ class _DefaultProcessUtils implements ProcessUtils {
     }
   }
 }
+
+/// Manages the creation of abstract processes.
+///
+/// Using instances of this class provides level of indirection from the static
+/// methods in the [Process] class, which in turn allows the underlying
+/// implementation to be mocked out or decorated for testing and debugging
+/// purposes.
+abstract class ProcessManager {
```

Comment:
Should all this go in a base/process_manager.dart ?

Reply:
I don't think its worth splitting up the libraries. Everyone that needs process.dart also needs process_manager.dart, and they'll likely have circular dependencies on eachother  if we start combining ProcessManager and ProcessUtils